includes/class-wc-payments-apple-pay-registration.php

```diff
@@ -61,15 +61,19 @@ class WC_Payments_Apple_Pay_Registration {
 	/**
 	 * Initialize class actions.
 	 *
-	 * @param WC_Payments_API_Client $payments_api_client WooCommerce Payments API client.
-	 * @param WC_Payments_Account    $account WooCommerce Payments account.
+	 * @param WC_Payments_API_Client   $payments_api_client WooCommerce Payments API client.
+	 * @param WC_Payments_Account      $account WooCommerce Payments account.
+	 * @param WC_Payment_Gateway_WCPay $gateway WooCommerce Payments gateway.
 	 */
-	public function __construct( WC_Payments_API_Client $payments_api_client, WC_Payments_Account $account ) {
+	public function __construct( WC_Payments_API_Client $payments_api_client, WC_Payments_Account $account, WC_Payment_Gateway_WCPay $gateway ) {
 		$this->domain_name             = $_SERVER['HTTP_HOST'] ?? str_replace( [ 'https://', 'http://' ], '', get_site_url() ); // @codingStandardsIgnoreLine
 		$this->apple_pay_verify_notice = '';
 		$this->payments_api_client     = $payments_api_client;
 		$this->account                 = $account;
+		$this->gateway                 = $gateway;
 
+		add_action( 'init', [ $this, 'add_domain_association_rewrite_rule' ], 5 );
```

Comment:
I see that this has been moved from `init` to its own action with a higher priority. Did the previous code not work well?

Reply:
It did, but the upgrade hook also runs during `init` at the same priority - 10, so we need to make sure the rewrite rules are added prior to that (just like they were when the entire code was in one method).

Reply:
I see, that makes perfect sense! Thanks for the explanation! 💯